### PR TITLE
modified database using transaction operations and aggregate functions

### PR DESCRIPTION
**Pull request by collinsmezie**
Add a column species of type string to the animal's table.
Implemented transactions.
Add queries for analyzing table data using aggregate functions.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -1,19 +1,69 @@
...
 
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3 OR weight_kg = 10.4 OR weight_kg = 17.3;
+-- SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+
+--TRANSACTION SECTION--
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
```
Comment (brytebee) [inline queries.sql:27]: - [ ] **Required:** You are required to verify that change was made before and after the rollback.

**_Hint:_** `SELECT * FROM animals;`

File: queries.sql [hunk 2]
```diff
@@ -1,19 +1,69 @@
...
+
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon%';
+UPDATE animals SET species = 'pokemon' WHERE species = 'unspecified';
+COMMIT;
```
Comment (brytebee) [inline queries.sql:34]: - [ ] **Required:** Same as above. Verify that the changes are made and persist before and after the `COMMIT`.

File: queries.sql [hunk 3]
```diff
@@ -1,19 +1,69 @@
...
+
+
+--TRANSACTION SECTION--
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon%';
+UPDATE animals SET species = 'pokemon' WHERE species = 'unspecified';
```
Comment (brytebee) [inline queries.sql:32]: - [ ] **Required:** Recall that you `ROLLBACK` the transaction which you set `...species = 'unspecified';`. So at this point where species isn't `digimon` would be `NULL`.

File: queries.sql [hunk 4]
```diff
@@ -1,19 +1,69 @@
...
+COMMIT;
+
+BEGIN TRANSACTION;
+DELETE * FROM animals;
+ROLLBACK;
```
Comment (brytebee) [inline queries.sql:41]: - [ ] **Required:** You are required to verify that the changes are rolled back after the rollback.

File: queries.sql [hunk 5]
```diff
@@ -1,19 +1,69 @@
...
-SELECT * FROM animals WHERE name NOT LIKE '%Gabumon';
+-- SELECT * FROM animals WHERE name NOT LIKE '%Gabumon';
 
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3 OR weight_kg = 10.4 OR weight_kg = 17.3;
+-- SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+
+--TRANSACTION SECTION--
+BEGIN TRANSACTION;
```
Comment (brytebee) [inline queries.sql:24]: - [ ] **Optional:** Favour the use of `BEGIN` over the old use of `BEGIN TRANSACTION`. They mean and do the same thing.

File: queries.sql [hunk 6]
```diff
@@ -1,19 +1,69 @@
...
+SELECT neutered, SUM(escape_attempts) FROM animals
+GROUP BY neutered;
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT species, MAX(weight_kg) MIN(weight_kg) FROM animals
+GROUP BY species;
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth
+BETWEEN '1990-01-01' AND '2000-01-01' GROUP BY species;
```
Comment (brytebee) [inline queries.sql]: - [ ] **Required:** Awesome 👏👏👏
However, using January 1st 2000 excludes the remaining months of the year 2000. Rather, use December 31 to cover the whole year.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @collinsmezie👋,

Good job so far coming so far with the progress you've made! 💪💪

## Highlights
You have done amazing in the following:
- Well documented and professionally looking project.✔️
- All required classes have been implemented.✔️
- Has implemented relationships between tables/classes.✔️
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (mateo951) [reply queries.sql:27]: Addressed ✔

File: queries.sql [hunk 7]
```diff
@@ -1,19 +1,77 @@
...
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon%';
+UPDATE animals SET species = 'pokemon' WHERE species = 'unspecified';
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
```
Comment (mateo951) [inline queries.sql:35]: Addressed ✔

File: queries.sql [hunk 8]
```diff
@@ -66,4 +74,4 @@ GROUP BY species;
 
 -- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
 SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth
-BETWEEN '1990-01-01' AND '2000-01-01' GROUP BY species;
+BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;
```
Comment (mateo951) [inline queries.sql:77]: Addressed ✔

**Review by mateo951 — APPROVED** [review]
### Hi @collinsmezie, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🔥 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**